### PR TITLE
enhance: remove CheckVecIndexWithDataTypeExist function in pkg and remove some cgo call

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/22837
related pr: https://github.com/milvus-io/milvus/pull/34104

### CODE REVIEW

**Comment by cqy123456** [comment]
rerun ut

File: internal/proxy/cgo_util.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"unsafe"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+)
+
+func CheckVecIndexWithDataTypeExist(name string, dType schemapb.DataType) bool {
+	cIndexName := C.CString(name)
+	cType := uint32(dType)
+	defer C.free(unsafe.Pointer(cIndexName))
```
Comment (jaime0815) [inline internal/proxy/cgo_util.go:35]: Can we check data type by Go code?  it is a little bit wired that proxy depends on segcore.

**Pull request merged**